sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/Transforms.java

```diff
@@ -40,7 +36,10 @@ private Transforms() {
      * @return The list holding all the generic items combined.
      */
     static <T> List<T> mapByIndex(Iterable<String> documents, BiFunction<String, String, T> mappingFunction) {
-        Objects.requireNonNull(documents, "'documents' cannot be null.");
+        // don't do anything if given input 'documents' is null, return null
```

Comment:
the documents is given by user. so it could be null. I will revert this check 
```
Objects.requireNonNull(documents, "'documents' cannot be null.");
```
in this method. And adding try-catch in API surface.